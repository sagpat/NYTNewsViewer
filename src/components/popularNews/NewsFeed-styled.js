import styled from 'styled-components';

export const Heading = styled.header`
    text-align: center;
    font-size: 30px;
    padding: 15px;
`;

export const ItemDiv = styled.header`
    width: 100%
`;

export const UnorderedList = styled.ul`
    list-style-type: none;
    margin: 1% 20%;
`;

export const ListedItem = styled.li`
    display: flex;
    padding: 10px;
    border-bottom: 1px solid #89ec26;
    cursor: pointer;
`;

export const HeadingLevelOne = styled.h1`
    font-size: 20px;
    font-weight: 400;
    margin-bottom: 0;
`;

export const Section = styled.section`
    display: flex;
`;


export const HeadingTwoAuthor = styled.h2`
    font-size: 16px;
    font-weight: 400;
`;

export const HeadingTwoDate = styled.h2`
    font-size: 16px;
    font-weight: 400;
    margin-left: auto;
    order: 2;
`;

export const Button = styled.button`
    padding-top: 2%;
    padding-left: 20px;
    color: red;
    font-size: 16px;
    font-weight: bold;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
`;