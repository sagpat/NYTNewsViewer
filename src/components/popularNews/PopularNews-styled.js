import styled from 'styled-components';

export const Heading = styled.header`
    text-align: center;
    font-size: 30px;
    padding: 15px;
`;

export const ItemDiv = styled.div`
    width: 100%
`;

export const UnorderedList = styled.ul`
    list-style-type: none;
    margin: 1% 20%;
`;

export const ListedItem = styled.li`
    display: flex;
    padding: 10px;
    border-bottom: 1px solid #89ec26;
`;

export const Image = styled.img`
    margin-right: 15px;
    border-radius: 50%;
`

export const TitleButton = styled.button`
    font-size: 17px;
    font-weight: 400;
    padding-left: 0;
    margin-bottom: 0;
    text-align: left;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
`;

export const SummaryParagraph = styled.p`
    padding-right: 5px;
`;

export const Section = styled.section`
    display: flex;
`;


export const HeadingTwoAuthor = styled.h2`
    font-size: 16px;
    font-weight: 400;
`;

export const HeadingTwoDate = styled.h2`
    font-size: 16px;
    font-weight: 400;
    margin-left: auto;
    order: 2;
`;

export const Button = styled.button`
    padding-top: 2%;
    padding-left: 20px;
    color: red;
    font-size: 16px;
    font-weight: bold;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
`;

export const NewsSection = styled.div`
    border: 1px solid #E0CA3C;
    padding: 5px;
    margin: auto;
    display: inline-block;
`;