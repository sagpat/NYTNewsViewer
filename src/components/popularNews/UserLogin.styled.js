import styled from 'styled-components';

export const LoginContainer = styled.div`
    margin: 20% 40%;
`;
export const LoginText = styled.h3`
`;

export const UserName = styled.input`
    padding: 0 2px 0 10px;
    height: 40px;
    width: 84%;
    box-sizing: border-box;
    border: 1px solid;
    margin-bottom: 15px;
`;

export const Password = styled.input`
    padding: 0 2px 0 10px;
    height: 40px;
    width: 84%;
    box-sizing: border-box;
    border: 1px solid;
    margin-bottom: 15px;
`;

export const Button = styled.button`
    padding: 10px 15px;
    display: block;
    margin-left: 60.5%;
    text-align: center;
    font-size: 16px;
    text-decoration: none;
    padding: 15px 20px;
    border: none;
    color: #fff;
    background-color: #000;
`;

export const Divider = styled.hr`
    border-color: #fff;
    boder-style: solid;
    boder-top: none;
    margin: 20px 0;
`;