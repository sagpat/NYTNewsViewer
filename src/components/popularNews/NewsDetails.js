import React from 'react';
import { SummaryParagraph, Section, HeadingTwoAuthor, HeadingTwoDate } from './PopularNews-styled';

const NewsDetails = (props) => {
    return (
        <React.Fragment>
            <SummaryParagraph>{props.summary}</SummaryParagraph>
            <Section>
                <HeadingTwoAuthor>{props.writtenBy ? props.writtenBy : '-'}</HeadingTwoAuthor>
                <HeadingTwoDate>Published Date: {props.publishedDate}</HeadingTwoDate>
            </Section>
        </React.Fragment>
    );
};

export default NewsDetails;
