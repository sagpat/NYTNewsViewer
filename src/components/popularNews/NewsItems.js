import React, { Component } from 'react';
import {
    ListedItem,
    ItemDiv,
    TitleButton,
    Button,
    Image
} from './PopularNews-styled';
import NewsDetails from './NewsDetails';

class NewsItems extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showNewsDetails: false
        }
    };

    toggleNewsDetails = () => {
        this.setState((prevState) => ({ showNewsDetails: !prevState.showNewsDetails }));
    }

    onClickShowCompleteNewsButton = (newsLink) => {
        window.open(newsLink, '_blank');
    }

    render() {
        const { url } = this.props.media[0]["media-metadata"][0];
        return (
            <ListedItem >
                <Image src={url} height="100px" width="100px" />
                <ItemDiv>
                    <TitleButton onClick={this.toggleNewsDetails}>{this.props.title}</TitleButton>
                    {this.state.showNewsDetails &&
                        <NewsDetails
                            summary={this.props.summary}
                            writtenBy={this.props.writtenBy}
                            publishedDate={this.props.publishedDate}
                            section={this.props.section}
                        />
                    }
                </ItemDiv>
                <Button onClick={() => this.onClickShowCompleteNewsButton(this.props.newsLink)}>></Button>
            </ListedItem>
        );
    }
};

export default NewsItems;