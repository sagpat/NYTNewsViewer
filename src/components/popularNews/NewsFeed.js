import React from 'react';
import { Heading, UnorderedList } from './PopularNews-styled';
import NewsItems from './NewsItems';

const NewsFeed = (props) => {
    return (
        <React.Fragment>
            <Heading>NYTimes Most Popular News Feed</Heading>
            {props.popularNewsData.map((data) => {
                return (
                    <UnorderedList key={data.views}>
                        <NewsItems
                            title={data.title}
                            summary={data.abstract}
                            writtenBy={data.byline}
                            publishedDate={data.published_date}
                            newsLink={data.url}
                            section={data.section}
                            media={data.media}
                        />
                    </UnorderedList>
                )
            })
            }
        </React.Fragment>
    );
};

export default NewsFeed;
