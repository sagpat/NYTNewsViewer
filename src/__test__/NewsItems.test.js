import React from 'react';
import { shallow } from 'enzyme';
import { ListedItem, ItemDiv, Button } from '../components/popularNews/PopularNews-styled';
import NewsItems from '../components/popularNews/NewsItems';

let wrapper;
const userProps = {
    summary: 'The little cat that escaped home',
    writtenBy: 'By Laura N',
    publishedDate: '12/05/2019',
    newsLink: 'abc.co.in',
    media: [
        {
            'media-metadata': [
                {
                    url: "https://static01.nyt.com/images/2019/06/21/us/politics/20dc-military1-sub/20dc-arms1-sub-square320.jpg"
                }
            ]
        }
    ]
};


beforeEach(() => {
    wrapper = shallow(<NewsItems
        summary={userProps.abstract}
        writtenBy={userProps.byline}
        publishedDate={userProps.published_date}
        newsLink={userProps.url}
        media={userProps.media} />);
});

describe('<NewsItems /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render one div', () => {
        expect(wrapper.find(ItemDiv)).toHaveLength(1);
    });

    it('should render one li element', () => {
        expect(wrapper.find(ListedItem)).toHaveLength(1);
    });


    it('should simulate Button click event', () => {
        global.open = jest.fn()
        wrapper.find(Button).simulate('click');
        expect(global.open.mock.calls.length).toEqual(1);
    });









});

