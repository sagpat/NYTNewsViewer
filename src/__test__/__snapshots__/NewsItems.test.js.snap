// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsItems /> rendering renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsItems
    media={
      Array [
        Object {
          "media-metadata": Array [
            Object {
              "url": "https://static01.nyt.com/images/2019/06/21/us/politics/20dc-military1-sub/20dc-arms1-sub-square320.jpg",
            },
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef(styled.img)
          height="100px"
          src="https://static01.nyt.com/images/2019/06/21/us/politics/20dc-military1-sub/20dc-arms1-sub-square320.jpg"
          width="100px"
        />,
        <ForwardRef(styled.div)>
          <ForwardRef(styled.button)
            onClick={[Function]}
          />
        </ForwardRef(styled.div)>,
        <ForwardRef(styled.button)
          onClick={[Function]}
        >
          &gt;
        </ForwardRef(styled.button)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "height": "100px",
          "src": "https://static01.nyt.com/images/2019/06/21/us/politics/20dc-military1-sub/20dc-arms1-sub-square320.jpg",
          "width": "100px",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
    margin-right: 15px;
    border-radius: 50%;
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(styled.button)
              onClick={[Function]}
            />,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": true,
                "rules": Array [
                  "
    font-size: 17px;
    font-weight: 400;
    padding-left: 0;
    margin-bottom: 0;
    text-align: left;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
",
                ],
              },
              "displayName": "styled.button",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "button",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          false,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    width: 100%
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": ">",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": ">",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": true,
            "rules": Array [
              "
    padding-top: 2%;
    padding-left: 20px;
    color: red;
    font-size: 16px;
    font-weight: bold;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
    display: flex;
    padding: 10px;
    border-bottom: 1px solid #89ec26;
",
        ],
      },
      "displayName": "styled.li",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "li",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(styled.img)
            height="100px"
            src="https://static01.nyt.com/images/2019/06/21/us/politics/20dc-military1-sub/20dc-arms1-sub-square320.jpg"
            width="100px"
          />,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.button)
              onClick={[Function]}
            />
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.button)
            onClick={[Function]}
          >
            &gt;
          </ForwardRef(styled.button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": "100px",
            "src": "https://static01.nyt.com/images/2019/06/21/us/politics/20dc-military1-sub/20dc-arms1-sub-square320.jpg",
            "width": "100px",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
    margin-right: 15px;
    border-radius: 50%;
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(styled.button)
                onClick={[Function]}
              />,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-EHOje",
                  "isStatic": true,
                  "rules": Array [
                    "
    font-size: 17px;
    font-weight: 400;
    padding-left: 0;
    margin-bottom: 0;
    text-align: left;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-EHOje",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            false,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    width: 100%
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": ">",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": ">",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-iwsKbI",
              "isStatic": true,
              "rules": Array [
                "
    padding-top: 2%;
    padding-left: 20px;
    color: red;
    font-size: 16px;
    font-weight: bold;
    border: none;
    background: none;
    outline: none;
    cursor: pointer;
",
              ],
            },
            "displayName": "styled.button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-iwsKbI",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
    display: flex;
    padding: 10px;
    border-bottom: 1px solid #89ec26;
",
          ],
        },
        "displayName": "styled.li",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "li",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
